this patch is derived from:

 From b642bce79e83a254e5af9b5c6619307ba22360c1 Mon Sep 17 00:00:00 2001
 From: Takuro Ashie <ashie@clear-code.com>
 Date: Tue, 13 Jun 2017 17:15:45 +0900
 Subject: [PATCH] Add a preference to force enable touch events without APZ
 
 It may be useful for touch devices.
 
 Signed-off-by: Takuro Ashie <ashie@clear-code.com>
 ---
  widget/gtk/nsWindow.cpp | 3 +++
  1 file changed, 3 insertions(+)

diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index ba4661b02a30..413fb707587a 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -515,6 +515,9 @@ nsWindow::nsWindow()
     }
 #endif
   }
+
+  if (Preferences::GetInt("dom.w3c_touch_events.enabled", 0) == 3)
+    RegisterTouchWindow();
 }
 
 nsWindow::~nsWindow() {
